src/odemis/util/img.py

```diff
@@ -811,3 +813,105 @@ def getBoundingBox(content):
+        """
+        self.steps = steps  # can be changed by the caller, on the fly
+        self._step = 0
+        self._img = None
+        self._best_dtype = None
+
+    def append(self, img):
+        """
+        Integrate two images (the new acquired image with the previous integrated one if exists) and return the
+        new integrated image. It will reset the ._img after reaching the number of integration counts, notifying
+        that the integration of the acquired images is completed.
+        Returns:
+            model.DataArray: the integrated image with the updated metadata
+        """
+        self._step += 1
+
+        if self._img is None:
+            orig_dtype = img[-1].dtype
+            self._best_dtype = get_best_dtype_for_acc(orig_dtype, self.steps)
+            self._img = img
+            return img
+        else:
+            # get the metadata from the last acquired image (should be the same for all images)
+            md = img[-1].metadata
+            logging.debug("Integrate one image after another")
+            data = numpy.add(img[0], img[1], dtype=self._best_dtype)
+            # update the metadata of the integrated image in every integration step
+            md = self.update_integration_metadada(md)
+            self._img = [model.DataArray(data, md)]
+
+            # At the end of the acquisition, check if the detector type is DT_NORMAL and then take the average by
+            # dividing with the number of acquired images (integration count) for every pixel position and restoring
+            # the original dtype.
+            if self._step == self.steps:
+                det_type = md.get(model.MD_DET_TYPE, model.MD_DT_INTEGRATING)
+                if det_type == model.MD_DT_NORMAL:  # SEM
+                    orig_dtype = img[-1].dtype
+                    data = numpy.divide(data, self.steps, dtype=orig_dtype)
+                if not det_type == model.MD_DT_INTEGRATING:  # optical
+                    logging.warning("Unknown detector type %s for image integration. "
+                                    "Image integration was performed anyways.", det_type)
+                # The baseline, if exists, should also be subtracted from the integrated image.
+                if model.MD_BASELINE in md:
+                    data, md = self.subtract_baseline(data, md)
+                # reset the ._img once you reach the integration count
+                self._img = None
+                self._step = 0
+
+            return [model.DataArray(data, md)]
+
+    def update_integration_metadada(self, md):
+        """
+        Update the metadata of each integrated image.
+        """
+        if MD_DWELL_TIME in md:
+            md[model.MD_DWELL_TIME] *= self._step  # total time ebeam stayed on same pixel/position
+        if MD_EXP_TIME in md:
+            md[model.MD_EXP_TIME] *= self._step
+        md[model.MD_INTEGRATION_COUNT] = self._step
```

Comment:
Yes, this works, because you pass the latest metadata, and normally all the images are acquired with the same settings. However I feel it'd be easier to understand and it would work in more cases if you were doing the addition of two metadata. I'd recommend something like that:
``` python
def add_integration_metadata(mda, mdb):
   """
   add mdb to mda, and update mda with the result
   returns dict: mda, which has been updated
   """
   if MD_DWELL_TIME in mda:
       mda[model.MD_DWELL_TIME] += mdb.get(model.MD_DWELL_TIME, 0)
   :
   mda[model.MD_INTEGRATION_COUNT] = mda.get(model.MD_INTEGRATION_COUNT, 1) + mdb.get(model.MD_INTEGRATION_COUNT,1)

   return mda
```